src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+            'name' => $clientName
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testUpdateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $iri = $this->findIriBy(Client::class, ['name' => 'client_2']);
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
+        $updatedName = $this->createClientName();
+        $updateClientJson = ClientMother::withAllParameters(
+            $updatedName,
+            1,
+            "description updated",
+            true,
+            5,
+            [$scriptId],
+            [],
+            -1,
+            "rsync long arguments updated",
+            "rsync short arguments updated",
+            "ssh arguments updated",
+            "root@172.0.0.2"
+        );
+        $httpClient->request('PUT', $iri, [
+            'json' => $updateClientJson
+        ]);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($updateClientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testUpdateClientInvalidMaxParallelJobs(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $iri = $this->findIriBy(Client::class, ['name' => 'client_3']);
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'description' => 'description updated',
+                'isActive' => true,
+                'maxParallelJobs' => -1,
+                'name' => $clientName,
+                'owner' => 1,
+                'quota' => -1,
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/ConstraintViolationList',
+            '@type' => 'ConstraintViolationList',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'maxParallelJobs: Max parallel jobs value must be a positive integer',
+        ]);
+    }
+    
+    public function testUpdateClientInvalidName(): void
+    {
+        $httpClient = $this->authenticate();
+        $iri = $this->findIriBy(Client::class, ['name' => 'client_4']);
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client_8',
+                'owner' => 1,
+                'quota' => -1
+            ]
+        ]);
```

Comment:
`
$existingClient = ClientMother::getExistingClient();
$existingName = $existingClient->getJson()[ClientMother::PARAM_NAME];

$client = ClientMother::withAllParameters();
$json = $client->getJson();
// Post to create the new client
$json[ClientMother::PARAM_NAME] = $existingName;
// PUT to update your client
`

In your way of doing you seem to use 2 existing clients which is great but then maybe you will need getExistingClient1 and getExistingClient2 in your ClientMother which is kind o... ummm. (To be honest ObjectMother's are thought to create instances not to obtain them but well... I guess this would good enough)

But it has a caveat, if somehow there is an error and the PUT succeds you end up with a well known client being updated and it might broke other tests, so creating a new one as I suggest has his own advantages too.